File: lib/feedback/ProgressIndicator.ts
export interface CircularProgressIndicatorProps {
    color?: string,
    strokeWidth?: number
}

export function CircularProgressIndicator(id: string, {
    color = "#0F0F0F",
    strokeWidth = 10
}: CircularProgressIndicatorProps){
    return {
        id,
        name: "CircularProgressIndicator",
        props: {
            color,
            strokeWidth
        }
    };
}

export interface LinearProgressIndicatorProps {
    color?: string,
    minHeight?: number
}

export function LinearProgressIndicator(id: string, {
    color=  "#000000",
    minHeight = 3
}: LinearProgressIndicatorProps){
    return {
        id,
        name: "LinearProgressIndicator",
        props: {
            color,
            minHeight
        }
    }
}

File: flutter_render/lib/src/feedback/ProgressIndicator.dart
import 'package:neneui_render/src/enum.dart';
import 'package:neneui_render/src/parser/Color.dart';
import 'package:shadcn_flutter/shadcn_flutter_experimental.dart';

// ignore: camel_case_types
class dCircularProgressIndicator {
  static Widget run({
    required BuildContext context,
    required Map<String, dynamic> data,
    required Function reRender,
    required Function event,
  }) {
    if (data['name'] == "CircularProgressIndicator") {
      event(Events.REGISTER_ID, {'id': data['id'], 'props': data['props']});

      return CircularProgressIndicator(
        color: ColorParse.parseColor(data['props']['color']),
        size: double.parse(data['props']['strokeWidth'].toString()),
      );
    } else {
      return SizedBox(child: Text("invalid"));
    }
  }
}

// ignore: camel_case_types
class dLinearProgressIndicator {
  static Widget run({
    required BuildContext context,
    required Map<String, dynamic> data,
    required Function reRender,
    required Function event,
  }) {
    if (data['name'] == "LinearProgressIndicator") {
      event(Events.REGISTER_ID, {'id': data['id'], 'props': data['props']});

      return LinearProgressIndicator(
        color: ColorParse.parseColor(data['props']['color']),
        minHeight: double.parse(data['props']['strokeWidth'].toString()),
      );
    } else {
      return SizedBox();
    }
  }
}
